src/common/gpio_motor.rs

```diff
@@ -437,3 +459,146 @@ where
         self.set_power(0.0)
     }
 }
+
+/// Represents a motor with an A and B pin. When moving forwards,
+/// a PWM signal is sent through the A pin and the B pin is set to high,
+/// vice versa for moving backwards. Note: If the dir_flip attribute is set to
+/// true, this functionality is reversed
+pub(crate) struct AbMotor<B> {
+    board: B,
+    a_pin: i32,
+    b_pin: i32,
+    max_rpm: f64,
+    dir_flip: bool,
+    is_on: bool,
+    pwm_pin: i32,
+}
+
+impl<B> AbMotor<B>
+where
+    B: Board,
+{
+    pub(crate) fn new(
+        a_pin: i32,
+        b_pin: i32,
+        max_rpm: f64,
+        dir_flip: bool,
+        board: B,
+    ) -> anyhow::Result<Self> {
+        let mut res = Self {
+            board,
+            a_pin,
+            b_pin,
+            max_rpm,
+            dir_flip,
+            is_on: false,
+            pwm_pin: a_pin,
+        };
+        // we start with this because we want to reserve a timer and PWM channel early
+        // for boards where these are a limited resource
+        res.board.set_pwm_frequency(a_pin, MOTOR_PWM_FREQUENCY)?;
+        res.board.set_pwm_duty(a_pin, 0.0)?;
```

Comment:
why not reserving a pwm pin for both?

Reply:
because we only ever need one PWM signal to run the motor (it's just that the pin the signal is running on changes), so I figured it was more economical to only reserve one timer